waddrmgr/scoped_manager.go

```diff
@@ -2055,13 +2117,19 @@ func (s *ScopedKeyManager) ImportScript(ns walletdb.ReadWriteBucket,
 		return nil, managerError(ErrCrypto, str, err)
 	}
 
+	// If a key isn't considered to be "secret", we encrypt it with the
+	// public key so we can create script addresses that also work in
+	// watch-only mode.
+	cryptoKey := s.rootManager.cryptoKeyScript
+	if !isSecretScript {
+		cryptoKey = s.rootManager.cryptoKeyPub
+	}
+
 	// Encrypt the script for storage in database using the crypto script
 	// key when not a watching-only address manager.
 	var encryptedScript []byte
-	if !s.rootManager.WatchOnly() {
-		encryptedScript, err = s.rootManager.cryptoKeyScript.Encrypt(
-			script,
-		)
+	if !s.rootManager.WatchOnly() || !isSecretScript {
```

Comment:
So then we don't encrypt at all if `isSecretScript` is `true`? In `scriptAddressRowToManaged`, we currently _always_ attempt to decrypt.

Reply:
Yeah, there was something backward. I fixed it. We now either require the manager to be unlocked (=cannot be watch-only) or the script to be public. And then we always encrypt, the question is just with what key.